### DISCUSSION TITLE
Not-a-Dev-Question: Loopback.IO the right framework?

### DISCUSSION THREAD

**Discussion by MarcelMertens** [loopbackio/loopback-next#8305] [Q&A]
Hello everyone,
first fact: I'm not a software developer, instead more a infrastructure / architect guy.
We are a MSP and hosting tons of workload in our own datacenter. Most workload is virtualized / containerized.
For many backend services we have own script and API which we use with several workflow and automation tools. Each backend service has it's own API syntax, or script parameters and there is no global authorization / permission system available.
Our idea is to abstract all of these APIs / scripts with a central globalized API with a authorization / permission system.

For example: Instead of use the PowerDNS API to an individual DNS server you send your request to the central API containing the customer, and dns specific information and the "Middleware" (Loopback.IO?) forwards this request to the responsible backend connector (Node.js script, phyton script etc). The Backend Connector speaks to the DNS server and response back to the Middleware.
In this a valid indented use of loopback? May someone can give me a hint how to integrate backend services. Your documentation is very powerful but i'm missing an overview about all the terms (maybe cause im not an dev).
Simplest way should be "https://loopback.io/doc/en/lb4/Calling-rest-apis.html". But not all backend services have ReST API or the API calls have to be transformed...

Kind Regards

Marcel

**Comment by mrmodise** [comment]
@MarcelMertens your use case sounds like mine. In our organisation servicing around a million customers a day we use LoopBack as an orchestration layer. We have microservices (mini apps) that are responsible for talking to the multiple backend systems to abstract their details from public facing applications with the flavour of centralised caching for improved performance, centralised logging for better reporting and security for safer apps for both the customer and business. I have found this works really well since LoopBack is built for API management. Developers can quickly get started with an API completing it in a matter of hours. The only challenge we had when were setting up was documentation so am glad you didnt face that due to the fact that it has improved over the years. We started using LoopBack while it was still under heavy development
